cpp/src/arrow/compute/api_scalar.h

```diff
@@ -278,6 +279,31 @@ struct ARROW_EXPORT DayOfWeekOptions : public FunctionOptions {
   uint32_t week_start;
 };
 
+/// Used to control timestamp timezone conversion and handling ambiguous/nonexistent
+/// times.
+struct ARROW_EXPORT AssumeTimezoneOptions : public FunctionOptions {
+ public:
+  /// How to interpret ambiguous local times that can be interpreted as
+  /// multiple instants due to DST shifts.
+  enum Ambiguous { AMBIGUOUS_RAISE, AMBIGUOUS_EARLIEST, AMBIGUOUS_LATEST };
+
+  /// How to handle local times that do not exists due to DST shifts.
+  enum Nonexistent { NONEXISTENT_RAISE, NONEXISTENT_EARLIEST, NONEXISTENT_LATEST };
+
+  explicit AssumeTimezoneOptions(std::string timezone,
+                                 Ambiguous ambiguous = AMBIGUOUS_RAISE,
+                                 Nonexistent nonexistent = NONEXISTENT_RAISE);
+  AssumeTimezoneOptions();
+  constexpr static char const kTypeName[] = "AssumeTimezoneOptions";
+
+  /// Timezone and timezone name to convert timestamp from
+  std::string timezone;
+  const arrow_vendored::date::time_zone* tz;
```

Comment:
I don't think this should be here. Is it deliberate or just a leftover?

Reply:
Oh. Probably a bad rebase.

Reply:
Removed.